### DISCUSSION TITLE
Specifying Boundary Conditions on Vector Field

### DISCUSSION THREAD

**Discussion by Vance-Turner** [zwicker-group/py-pde#220] [Q&A] [answered]
Hello,
Thank-you for this excellent project and using type hints throughout the code! I am trying to specify boundary conditions on a vector field where the boundaries are specific to the component. This is for a simple lid-driven cavity flow; so I want velocity on all the boundaries to be zero except for the tangential component on one boundary. I thought this could be accomplished with the "normal_value" key:

```
grid = CartesianGrid([[0,2.0],[0,2.0]], 41)
vF = VectorField(grid,data=np.zeros((2,41,41)))
vF.gradient(bc=[
    [{"normal_value":0,"value":0},{"normal_value":0,"value":0}],
    [{"normal_value":0,"value":0},{"normal_value":0,"value":1}]
]).data[0]
```
The example above is a just a test to see if I can set boundary conditions on a vector field. 
I get the following error:

``BCDataError: Boundary conditions `['normal_value', 'value']` are not supported.``

(Edited to make gradient of the vector field itself. Not the dot product!)

**Comment by david-zwicker** [comment, marked as answer]
It might be best to specify the vectorial boundary conditions with all components explicitly, like so

```
bc_x = [{"value": [0, 0]}, {"value": [0, 0]}]
bc_y = [{"value": [0, 0]}, {"value": [1, 0]}]
vF.gradient(bc=[bc_x, bc_y]).plot()
```

**Reply by Vance-Turner** [reply]
Thank-you for the quick reply! That makes sense.
